common/djangoapps/track/middleware.py

```diff
@@ -142,6 +143,10 @@ def enter_request_context(self, request):
         for header_name, context_key in META_KEY_TO_CONTEXT_KEY.iteritems():
             context[context_key] = request.META.get(header_name, '')
 
+        # HTTP_USER_AGENT user might can contain the information that include latin1 characters
+        # decoding this using latin1 scheme will prevent to raise UnicodeDecodeError when using
+        # json.dumps for tracking purpose.
+        context[HTTP_USER_AGENT_KEY] = context[HTTP_USER_AGENT_KEY].decode('latin1')
```

Comment:
We will also have to make this change here: https://github.com/edx/edx-platform/blob/master/common/djangoapps/track/views/__init__.py#L108

There is some sizable tech debt here (two different event emission paths). (sorry!)